### PR TITLE
XWIKI-15833: Users shouldn't be able to pick a group as a subgroup of itself

### PR DESCRIPTION
**Pull request by manuelleduc**
- Fixes the use of invalid tags in the javadoc.

https://jira.xwiki.org/browse/XWIKI-15833

### CODE REVIEW

File: xwiki-platform-core/xwiki-platform-user/xwiki-platform-user-script/src/main/java/org/xwiki/user/script/GroupScriptService.java [hunk 1]
```diff
@@ -147,16 +147,16 @@
...
 
     /**
-     * Checks if a group (candidate) can be added to the members of another group (target).<br />
+     * Checks if a group (candidate) can be added to the members of another group (target).<br>
      * A group (candidate) can be added to the members of another group (target) if the target is not, transitively,
-     * already part of the members of candidate.<br />
-     * A group cannot be added to its own members.<br />
-     * A group (candidate) cannot be added to the members of a group (target) if candidate is already one of the direct
-     * members of target.<br />
+     * already part of the members of the candidate.<br>
```
Comment (surli) [inline xwiki-platform-core/xwiki-platform-user/xwiki-platform-user-script/src/main/java/org/xwiki/user/script/GroupScriptService.java:152]: that don't fix the issue at all. Here the problem is about `<` and `>` which are not encoded.

Comment (surli) [inline xwiki-platform-core/xwiki-platform-user/xwiki-platform-user-script/src/main/java/org/xwiki/user/script/GroupScriptService.java:152]: why don't you just remove those br tags? they are not really useful are they?

Comment (vmassol) [reply xwiki-platform-core/xwiki-platform-user/xwiki-platform-user-script/src/main/java/org/xwiki/user/script/GroupScriptService.java:152]: if you need a list you can use `<ul>` and `<li>`

Comment (tmortagne) [reply xwiki-platform-core/xwiki-platform-user/xwiki-platform-user-script/src/main/java/org/xwiki/user/script/GroupScriptService.java:152]: In general think of Javadoc as a piece of HTML with a few helpers based on the `{@something}` syntax.

Comment (vmassol) [reply xwiki-platform-core/xwiki-platform-user/xwiki-platform-user-script/src/main/java/org/xwiki/user/script/GroupScriptService.java]: It's not HTML. You don't close elements for example and only a small subset is supported.

BR should never be used. If you need items you should use a list.

Comment (manuelleduc) [reply xwiki-platform-core/xwiki-platform-user/xwiki-platform-user-script/src/main/java/org/xwiki/user/script/GroupScriptService.java]: Javadoc updated with ul/li tags. For now on I'll add a `mvn javadoc:javadoc` to my pre-push tests.

Comment (tmortagne) [reply xwiki-platform-core/xwiki-platform-user/xwiki-platform-user-script/src/main/java/org/xwiki/user/script/GroupScriptService.java]: > BR should never be used. If you need items you should use a list.

My point is that to would not do a list using `*` and `<br/>` in HTML.

**Pull request merged**